src/chips/hotel/src/chip.rs

```diff
@@ -27,11 +28,19 @@ impl Chip for Hotel {
 
     fn service_pending_interrupts(&mut self) {
         unsafe {
-            cortexm3::nvic::next_pending().map(|nvic_num| {
-                match nvic_num {
-                    _   => {}
+            loop {
+                match cortexm3::nvic::next_pending() {
+                    Some(nvic_num) => {
```

Comment:
Ah, good catch. I had actually done this change locally and it worked out ok, but what it's actually doing is iterating over the one `Option` returned by `next_pending()`, which yields either 1 or 0 elements. Oops.

You can use [while let](http://rustbyexample.com/flow_control/while_let.html) instead, though.